startupscript/butane/006-start-proxy-agent.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# start-proxy-agent.sh starts the proxy agent on the VM.
+# Note: This scripts requires agent specific environment to be set in /home/core/agent.env on the VM.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+if [[ $# -ne 1 ]]; then
+    echo "usage: $0 <proxyImage>"
+    exit 1
+fi
+
+readonly PROXY_IMAGE="$1"
+port=$(docker inspect application-server | jq -r '.[].NetworkSettings.Ports | to_entries[] | .value[] | select(.HostIp == "0.0.0.0" or .HostIp == "::") | .HostPort' | head -n 1)
+echo "got port: ${port}"
+if [[ -z "${port}" ]]; then
```

Comment:
Since the if condition exits, this is cleaner as:

```
if [[ -z "${PORT}" ]]; then
    echo "Error: Port is empty."
    exit 1
fi

echo "Agent port is ${PORT}"
```